cmake/ace_group.cmake

```
@@ -63,7 +63,9 @@ function(_opendds_vs_force_static)
   endif()
 endfunction()
 
-if(MSVC AND OPENDDS_STATIC)
+if(MSVC AND OPENDDS_STATIC AND DEFINED OPENDDS_STATIC_RUNTIME AND (OPENDDS_STATIC_RUNTIME STREQUAL "" OR OPENDDS_STATIC_RUNTIME))
```

It looks like `OPENDDS_STATIC_RUNTIME` will always be set in the generated config.cmake, is that correct? If so, I think this line can skip the `DEFINED` check.

---

The check is required since this code is used for cases that don't have a `config.cmake` (these are cases where OpenDDS is built with CMake).

---

Yes, we do need it work without it being defined, but the way it's done here changes the behavior of the CMake-built OpenDDS. Now it won't ever force it by default.... which might be fine since it now leaves it to CMake to decide. Is that intentional?

I'm also still not comfortable with how the empty string is being used as a pseudo-true value. If we're assuming the current behavior, this could be:

```suggestion
if(MSVC AND OPENDDS_STATIC AND DEFINED OPENDDS_STATIC_RUNTIME AND OPENDDS_STATIC_RUNTIME)
```

Then the default value in the configure script would be changed to `1`.

---

The `configure` script is reporting one of three states, it's not boolean.  Empty, ON, and OFF are each their own values.

Maybe that's a non-optimal design but I'm not that interested in fixing it in this PR.

---

> The `configure` script is reporting one of three states, it's not boolean. Empty, ON, and OFF are each their own values.

> I guess that didn't really answer the question -- we're also trying to determine if the configure script was used at all.

Okay, but the code doesn't distinguish between `OPENDDS_STATIC_RUNTIME` being empty and `OPENDDS_STATIC_RUNTIME` being true, so there's no point to breaking a standard convention here.

---

Empty being treated as true is only the case when `MSVC` is also true.  Other platforms have their own defaults for static vs. shared linking to standard library runtime libraries.

The `OPENDDS_STATIC_RUNTIME` value is just relaying to CMake how `configure` was run (user specified `--static-runtime`, user specified `--no-static-runtime`, or neither), it's not adding additional logic for defaults or specific platforms.

---

If we want leave it open to cover those cases, we could leave it undefined if it wasn't specified, then:

```suggestion
if(MSVC AND OPENDDS_STATIC)
  if(NOT DEFINED OPENDDS_STATIC_RUNTIME AND NOT (_OPENDDS_CMAKE_BUILT_AND_INSTALLED OR OPENDDS_IS_BEING_BUILT))
    set(OPENDDS_STATIC_RUNTIME TRUE)
  endif()
  if(OPENDDS_STATIC_RUNTIME)
```

---

That would require more changes to `configure`, and I don't see any cases where it already handles an `undef` Perl value or has other logic for omitting it from `config.cmake`.